prompt2model/utils/parse_json_responses.py

```diff
@@ -31,19 +31,25 @@ def extract_response(
     try:
         response_json = json.loads(response_text, strict=False)
     except json.decoder.JSONDecodeError:
+        # TODO: Should error be thrown here?
         logger.warning(f"API response was not a valid JSON: {response_text}")
         return None
 
     missing_keys = [key for key in required_keys if key not in response_json]
     if len(missing_keys) != 0:
+        # TODO: Should error be thrown here?
```

Comment:
Probably no error should be thrown in either of these cases; if the return value is None, then we should consider either retrying the request or using some default behavior (e.g. ignoring the dataset)

Reply:
So currently it returns None, and that causes it to retry the request - till max API retry count is reached. However, I don't really think retrying the request would change much since our temperature is practically 0. Probably makes sense to the signal approach.